File: backend/semgrep_rules/tests/java/kisa_sw_001_bad.java
import java.sql.Statement;
import javax.servlet.http.HttpServletRequest;

class SqlInjectionBad {

    public void search(
        HttpServletRequest request,
        Statement statement
    ) throws Exception {

        String id =
            request.getParameter("id");

        String query =
            "SELECT * FROM users WHERE id = "
            + id;

        statement.executeQuery(
            query
        );
    }
}

File: backend/semgrep_rules/tests/python/kisa_sw_001_bad.py
import sqlite3

from django.http import HttpRequest


def find_user(
    request: HttpRequest
):

    user_id = (
        request.GET.get(
            "id"
        )
    )


    connection = (
        sqlite3.connect(
            "test.db"
        )
    )


    cursor = (
        connection.cursor()
    )


    query = (
        "SELECT * "
        "FROM users "
        "WHERE id = '"
        + user_id
        + "'"
    )


    cursor.execute(
        query
    )


    result = (
        cursor.fetchall()
    )


    connection.close()


    return result